lib/_http_server.js

```diff
@@ -278,7 +279,12 @@ function writeHead(statusCode, reason, obj) {
   if (this[kOutHeaders]) {
     // Slow-case: when progressive API and header fields are passed.
     let k;
-    if (obj) {
+    if (ArrayIsArray(obj)) {
```

Comment:
should we at least verify that `obj.length` is a multiple of two?

Reply:
What should we do it it's not? Throw?

Reply:
I would throw, yes

Reply:
Fixed